src/quo2/components/navigation/page_nav/view.cljs

```diff
@@ -0,0 +1,285 @@
+(ns quo2.components.navigation.page-nav.view
+  (:require [quo2.components.avatars.group-avatar.view :as group-avatar]
+            [quo2.components.buttons.button.view :as button]
+            [quo2.components.dropdowns.dropdown :as dropdown]
+            [quo2.components.icon :as icons]
+            [quo2.components.markdown.text :as text]
+            [quo2.components.navigation.page-nav.style :as style]
+            [quo2.theme :as theme]
+            [react-native.core :as rn]))
+
+(def ^:private button-type
+  {:white       :grey
+   :neutral-5   :dark-grey
+   :neutral-90  :grey
+   :neutral-95  :dark-grey
+   :neutral-100 :black
+   :photo       :grey
+   :blur        :grey})
+
+(defn- page-nav-base
+  [{:keys [margin-top background on-press accessibility-label icon-name]
+    :or   {background :white}}
+   & children]
+  (into [rn/view {:style (style/container margin-top)}
+         (when icon-name
+           [button/button
+            {:type                (button-type background)
+             :icon-only?          true
+             :size                32
+             :on-press            on-press
+             :accessibility-label accessibility-label
+             :background          (when (#{:photo :blur} background) background)}
+            icon-name])]
+        children))
+
+(defn- right-section-spacing [] [rn/view {:style style/right-actions-spacing}])
+
+(defn- add-right-buttons-xf
+  [max-actions background]
+  (comp (filter map?)
```

Comment:
Excellent usage of transducers!

Reply:
Any good resources on transducers? :)

Reply:
@J-Son89  
The official clojure docs:

https://clojure.org/reference/transducers 

particularly the section: *Defining Transformations With Transducers*
and I'm using the `into` version: https://clojure.org/reference/transducers#_into